src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs

```diff
@@ -29,7 +30,23 @@ public static AzureAuthenticationInfo GetConfiguredAzureAuthentication(IConfigur
                 authenticationConfiguration = new AuthenticationConfiguration();
             }
 
-            var applicationKey = configuration.GetValue<string>(EnvironmentVariables.Authentication.ApplicationKey);
+            string applicationKey;
+
+            if (!string.IsNullOrEmpty(authenticationConfiguration.SecretFilePath) && !string.IsNullOrEmpty(authenticationConfiguration.SecretFileName))
```

Comment:
Can you verify this is blocked in startup validation please? If so, can you share the full log to see what the experience looks like?

Reply:
So if I start it without a key in env var and file; does it pass?

Reply:
Yes until a call to GetConfiguredAzureAuthentication is made. like it is now with the current configuration.
A check of identityId and applicationKey are being made on the AzureAuthenticationFactory, not by a vlidation step

Reply:
No problem; I've opened #2187 for this.